folio-spring-base/src/test/java/org/folio/spring/client/EnrichUrlAndHeadersClientTest.java

```diff
@@ -0,0 +1,201 @@
+
+    FolioExecutionContext context = mock(FolioExecutionContext.class);
+    when(context.getOkapiHeaders()).thenReturn(Map.of("z", List.of("z-val")));
+    when(context.getAllHeaders())
+      .thenReturn(
+        Map.of(
+          "misc-1",
+          List.of("misc-1-val"),
+          "accept-language",
+          List.of("en-US,en;q=0.9"),
+          "misc-2",
+          List.of("misc-2-val")
+        )
+      );
+
+    assertThat(
+      EnrichUrlAndHeadersClient.prepareHeaders(request, context),
+      is(
+        allOf(
+          hasEntry("a", List.of("a-val")),
+          hasEntry("b", List.of("b-val")),
+          hasEntry("c", List.of("c-val")),
+          hasEntry("z", List.of("z-val")),
+          hasEntry("Accept-Language", List.of("en-US,en;q=0.9")),
+          is(aMapWithSize(5))
+        )
+      )
+    );
+  }
+
+  @Test
+  void testRequestExecution() throws IOException {
+    final Request request = Request.create(
+      Request.HttpMethod.GET,
+      "http://test-url",
+      Map.of("a", List.of("a-val")),
+      Request.Body.create("test-data"),
+      null
+    );
+
+    final FolioExecutionContext context = mock(FolioExecutionContext.class);
+    when(context.getOkapiUrl()).thenReturn("http://okapi");
+    when(context.getOkapiHeaders()).thenReturn(Map.of("z", List.of("z-val")));
+    when(context.getAllHeaders())
+      .thenReturn(
+        Map.of(
+          "misc-1",
+          List.of("misc-1-val"),
+          "accept-language",
+          List.of("en-US,en;q=0.9"),
+          "misc-2",
+          List.of("misc-2-val")
+        )
+      );
+
+    final Request.Options options = new Request.Options();
+
+    final EnrichUrlAndHeadersClient client = new EnrichUrlAndHeadersClient(
+      context,
+      null
+    );
+
+    // best way to check for this without having to do a bunch of feign/okhttp logic
+    final OkHttpClient mockedDelegate = mock(OkHttpClient.class);
+    ReflectionTestUtils.setField(client, "delegate", mockedDelegate);
+
+    when(mockedDelegate.execute(any(Request.class), eq(options)))
+      .thenReturn(Response.builder().request(request).status(299).build());
+
+    // ensure response is returned upon execution
+    assertThat(client.execute(request, options).status(), is(299));
+
+    // verify that the correct request was sent with the correct options
+    ArgumentCaptor<Request> requestCaptor = ArgumentCaptor.forClass(
+      Request.class
+    );
+    verify(mockedDelegate, times(1))
+      .execute(requestCaptor.capture(), eq(options));
+
+    assertThat(
+      requestCaptor.getValue().httpMethod(),
+      is(Request.HttpMethod.GET)
+    );
+    assertThat(requestCaptor.getValue().url(), is("http://okapi/test-url"));
+    assertThat(
+      requestCaptor.getValue().headers(),
+      is(
+        allOf(
+          hasEntry("a", List.of("a-val")),
+          hasEntry("z", List.of("z-val")),
+          hasEntry("Accept-Language", List.of("en-US,en;q=0.9")),
+          is(aMapWithSize(3))
```

Comment:
Hamcrest's map comparator requires keys be in the same order, which may not be the case (encountered this with the one including accept-language). Rather than have a test rely on this, I opted to go with this solution.